### DISCUSSION TITLE
Three problems

### DISCUSSION THREAD

**Discussion by DynamicLoader** [Pconti31/TTF2GFX#5] [Q&A] [answered]
### Problem 1
My %JAVA_HOME% is "C:\Program Files\...",which is default to Windows Java installation.
There is a space between "Program" and "Files",in that case the bat file
```Bat
start "" /min %JAVA_HOME%\bin\javaw -jar ttf2gfx-1.00.jar
```
will not work.
I added quotation marks like this `start "" /min "%JAVA_HOME%\bin\javaw" -jar ttf2gfx-1.00.jar` and it works.

### Problem 2
By trying the application,I found the table on the left doesn't have a scroll bar,which means it can't be scrolled.
In that case,if end code is set to large (e.g. 2000),preview is not fully showed.
[**Edit**] It seems that the GFX does not accept chars that bigger then 255?

### Problem 3
Only when suffix of file is ".ttf",the [Generate] popup is showed.
But when I use ".otf" or even ".ttc",it can load successfully but the [Generate] popup is NOT showed.
After I change their suffix to ".ttf",it popuped normally and generate successfully.

**Comment by DynamicLoader** [comment]
Solved.

**Comment by Pconti31** [comment, marked as answer]
@Lucy2003 
Problem 1:  I'll fix the bat file.  
Problem 2:  Yes, Adafruit GFX fonts are 7 bit so 127 is normally the largest character as an integer but some people like me have successfully used 255 as maximum.  Old PCs had 8 bit code sets like Latin1-Latin15.  No idea what China used.
Problem 3: Only ttf fonts show popup but .otc and ,ttc do load.  Hmmm, I had no idea.  I mean the tool is named ttf to gfx so I built it to only expect ttf fonts.  Seems Java supports reading .otc (no sure what .ttc is).  I'll examine the code and fix bringing up the popup when these files are read.  
Thanks for pointing these issues out.
Paul--
